templates/new/mix.exs

```diff
@@ -75,7 +75,8 @@ defmodule <%= app_module %>.MixProject do
   defp aliases(_target) do
     [
       "deps.precompile": ["nerves.precompile", "deps.precompile"],
-      "deps.loadpaths": ["deps.loadpaths", "nerves.loadpaths"]
+      "deps.loadpaths": ["deps.loadpaths", "nerves.loadpaths"],
+      "deps.get": ["deps.get", "nerves.deps.get"]
```

Comment:
What happens to people with existing projects that don't alias `deps.get`?

Also, as an alternative, what do you think of changing this whole thing to:
```
defp aliases(_target), do: Nerves.Bootstrap.default_aliases()
```
or
```
defp aliases(_target) do
  [
    "deps.precompile": ["mycustom.precompile", "deps.precompile"],
  ] |> Nerves.Bootstrap.add_aliases()
end
```
The main idea being that users don't have to update their `mix.exs` files whenever this changes. Also, can the `default_aliases/0` function be smart and see what version of `nerves` is being used so that it knows whether or not to include the `nerves.deps.get` alias?

Reply:
I don't believe that there is anything that we need to handle in nerves_bootstrap regarding existing projects that do not declare the alias for `deps.get`. The only thing from this standpoint that they would be missing out on is check for update. We should discuss this further over in https://github.com/nerves-project/nerves/pull/235 because it has a bigger impact there.

As far as how to handle the aliases:
Unfortunately we are stuck in a catch 22 when it comes to simplifying the mix file. We can't depend on code from archives because if the archive is not installed, it would fail in a way that doesn't provide a helpful error. In addition, we can't really hide the aliases that are being applied because if the developer would want to add an alias to one that we are already overriding, mix wouldn't know how to reconcile them. Its best that this remains explicit.

Reply:
After typing up a response in the other PR, I realized that we _would_ have to handle the update warning here. Lets put it in `loadpaths`

Reply:
A couple points:
1. Aren't existing projects missing out on more than the update check if they don't add the Nerves alias for `deps.get`? If that's the case, we should move the update check to a different alias and skip out on this. However, I don't believe that to be the case. I think that it's really mandatory for everyone to update their projects.
2. I think that the word "explicit" is being misused to justify adding implementation details to everyone's mix.exs files. I'll be honest that I don't understand why Nerves needs to plug in at precompile and loadpaths. I haven't tried removing those lines, but I'm pretty sure that would break things. There's no documentation on how they must be ordered or what they do, so I'm not sure how any user would be able to do anything with them.
3. Regarding mix not being able to reconcile differences in aliases, I'm not proposing that. See the examples above.
4. If giving good error messages is a concern, I suggest adding a one-liner to everyone's mix.exs to check if `nerves_bootstrap` is available.